modules/gapi/src/backends/cpu/gcpubackend.cpp

```diff
@@ -199,7 +241,14 @@ void cv::gimpl::GCPUExecutable::run(std::vector<InObj>  &&input_objs,
         }
 
         // Now trigger the executable unit
-        k.apply(context);
+        if (k.m_isStateful)
+        {
+            k.m_stRunF(context, m_nodesToStates[op_info.nh]);
+        }
+        else
+        {
+            k.m_runF(context);
+        }
```

Comment:
Instead, you can optionally put a state into context and call the same old entry point.

Reply:
This was done previously when I tried to hide all information for communication between GCPUExecutable and GCPUKernel in GCPUContext.
The reason why I haven't just hide state in GCPUContext is in inconsistent(only from my personal opinion) GCPUKernel interface - ```m_runF``` callback hides all information in GCPUContext for purpose of making stateless kernel call and statefull kernel call similar, when ```m_setupF```  shows all internals (or part of internals - GArgs) out.

That's why I want either to hide all information in contexts or expose all information outside.
I've chosen last option because contexts bring more code lines to fill them.
Also if I will think from GCPUExecutable point of view, then I:
1) accept graph of nodes, where each op node knows whether it is statefull or not;
2) if node is stateful it creates and holds state for it;
3) during run it checks if node is stateful and calls stateful callback
My brain found it as transparent contract between GCPUBackend and GCPUKernel where GCPUBackend doesn't know about some GCPUKernel inner details likes a requirement to fill or not to fill GCPUContext with state.

Reply:
The last time you did it wrong, since you reused the same context object for two different types of callbacks (a compile-time one and a run-time one).

Reply:
Done.